sdk/core/azure-core-management/src/main/java/com/azure/core/management/implementation/polling/SyncPollOperation.java

```diff
@@ -0,0 +1,170 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.management.implementation.polling;
+
+import com.azure.core.http.HttpHeaders;
+import com.azure.core.http.HttpMethod;
+import com.azure.core.http.HttpPipeline;
+import com.azure.core.http.HttpRequest;
+import com.azure.core.http.HttpResponse;
+import com.azure.core.http.rest.Response;
+import com.azure.core.management.polling.PollResult;
+import com.azure.core.util.BinaryData;
+import com.azure.core.util.Context;
+import com.azure.core.util.polling.LongRunningOperationStatus;
+import com.azure.core.util.polling.PollResponse;
+import com.azure.core.util.polling.PollingContext;
+import com.azure.core.util.serializer.SerializerAdapter;
+
+import java.lang.reflect.Type;
+import java.time.Duration;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+/**
+ * Synchronous poll operation for Azure resource manager (ARM) long-running-operation (LRO).
+ */
+public class SyncPollOperation {
+    private static final LongRunningOperationStatus LRO_CANCELLED
+        = LongRunningOperationStatus.fromString("Cancelled", true);
+
+    /**
+     * Gets a Function that starts the Azure resource manager(ARM) long-running-operation(LRO).
+     *
+     * @param serializerAdapter the serializer for any encoding and decoding
+     * @param pollResultType the type of the poll result, if no result is expecting then this should be Void.class
+     * @param lroInitialResponse the activation operation to activate (start) the long-running operation
+     * @param <T> the type of poll result
+     * @return the ARM LRO activation Function
+     */
+    public static <T> Function<PollingContext<PollResult<T>>, PollResponse<PollResult<T>>> syncActivationFunction(
+        SerializerAdapter serializerAdapter, Class<T> pollResultType,
+        Supplier<Response<BinaryData>> lroInitialResponse) {
```

Comment:
nit, may easier to understand, if name it `lroInitialResponseSupplier`, as this is not the response itself.

Reply:
changed